pkg/spi-shared/secretstorage/secretstorage.go

```diff
@@ -47,3 +50,112 @@ type SecretStorage interface {
 	// Delete deletes the data of given id. A NotFoundError is returned if there is no such data.
 	Delete(ctx context.Context, id SecretID) error
 }
+
+// TypedSecretStorage is a generic "companion" to the "raw" SecretStorage interface which uses
+// strongly typed arguments instead of the generic SecretID and []byte.
+type TypedSecretStorage[ID any, D any] interface {
+	// Initialize initializes the connection to the underlying data store, etc.
+	Initialize(ctx context.Context) error
+	// Store stores the provided data under given id
+	Store(ctx context.Context, id *ID, data *D) error
+	// Get retrieves the data under the given id. A NotFoundError is returned if the data is not found.
+	Get(ctx context.Context, id *ID) (*D, error)
+	// Delete deletes the data of given id. A NotFoundError is returned if there is no such data.
+	Delete(ctx context.Context, id *ID) error
+}
+
+// DefaultTypedSecretStorate is the default implementation of the TypedSecretStorage interface
+// that uses the provided functions to convert between the id and data types to SecretID and []byte
+// respectively.
+type DefaultTypedSecretStorate[ID any, D any] struct {
```

Comment:
can you please write tests for `DefaultTypedSecretStorate`

Reply:
also I'm not sure about the name `Default`, but we can rename later when we have more implementations and know more